tests/trainer/test_trainer.py

```diff
@@ -2097,6 +2096,23 @@ def training_step(self, batch, batch_idx):
             2,
         ),
         ({"strategy": DDPShardedStrategy(), "gpus": 2}, DDPShardedStrategy, "ddp_sharded", GPUAccelerator, 2),
+        ({"strategy": "ddp2", "gpus": 2, "num_nodes": 2}, DDP2Strategy, "ddp2", GPUAccelerator, 2),
```

Comment:
Thanks for the additional test cases. 
I think in general we should probably also add a separate test for setting the value on any strategy instance.